system/Commands/Generators/CommandGenerator.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Generators;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\CLI\GeneratorTrait;
+
+/**
+ * Generates a skeleton command file.
+ */
+class CommandGenerator extends BaseCommand
+{
+	use GeneratorTrait;
+
+	/**
+	 * The Command's group.
+	 *
+	 * @var string
+	 */
+	protected $group = 'Generators';
+
+	/**
+	 * The Command's name
+	 *
+	 * @var string
+	 */
+	protected $name = 'make:command';
+
+	/**
+	 * The Command's short description
+	 *
+	 * @var string
+	 */
+	protected $description = 'Generates a new spark command.';
+
+	/**
+	 * The Command's usage
+	 *
+	 * @var string
+	 */
+	protected $usage = 'make:command <name> [options]';
+
+	/**
+	 * The Command's arguments
+	 *
+	 * @var array
+	 */
+	protected $arguments = [
+		'name' => 'The command class name.',
+	];
+
+	/**
+	 * The Command's options
+	 *
+	 * @var array
+	 */
+	protected $options = [
+		'--command'   => 'The command name, Defaults: "command:name"',
+		'--group'     => 'The command group, Defaults: [basic -> "CodeIgniter", generator -> "Generators"].',
+		'--type'      => 'The command type. Options [basic, generator], Defaults: "basic".',
+		'--namespace' => 'Set root namespace. Defaults: "APP_NAMESPACE".',
+		'--force'     => 'Force overwrite existing file.',
+		'--prevent'   => 'Prevent renaming class as framework standards format (e.g. UserCommand => User).',
```

Comment:
I don't think we have that convention strictly followed with the framework itself. Take a look at current CLI commands. Those do not mostly have `Command` suffixes. For others (controllers, migrations, etc.), there is no convention either for suffixing the component name. For controllers if you are using `$routes->setAutoRoute(true);` you would definitely not suffix `Controller` in the class name.

For this and other commands, I propose that suffixing the component name is default to `false`. If users would want their classes suffixed, then add the option to the command call. I guess from `--prevent` to `--suffix` to add the suffix.

Reply:
--suffix brilliant suggest, Thank you